pkg/storage/s3/s3.go

```diff
@@ -86,15 +91,31 @@ func NewImageStore(rootDir string, cacheDir string, gc bool, gcDelay time.Durati
 		linter:           linter,
 	}
 
-	cachePath := path.Join(cacheDir, CacheDBName+storage.DBExtensionName)
+	cachePath := path.Join(cacheDir, CacheDBName+constants.DBExtensionName)
 
 	if dedupe {
-		imgStore.cache = storage.NewCache(cacheDir, CacheDBName, false, log)
+		if len(cacheDriver) != 0 {
```

Comment:
Let's not do this magic. Pass a single valid initialized cacheDriver in the function parameters.
There are too many fallbacks to boltdb in this PR.